### PR TITLE
Intial draft for frontend review doc

### PR DESCRIPTION
**Pull request by hugohaggmark**
Is probably part of #15990?

- tried to keep it front end specific but there's probably stuff here that could be reused as general review checks
- is there something large missing?

### CODE REVIEW

**Comment by aocenas** [comment]
I think some of the more specific parts about code style (like using enums/const or bad practice examples) should be part of the https://github.com/grafana/grafana/blob/master/style_guides/frontend.md otherwise there will be a lot of overlaps. I think this can be short and having
> The pull request adds value and the impact of the change is in line with [Frontend Style Guide](https://github.com/grafana/grafana/blob/master/style_guides/frontend.md).

should cover all the code style checks without duplication. Otherwise I think it makes lots of sense.

**Comment by hugohaggmark** [comment]
> I think some of the more specific parts about code style (like using enums/const or bad practice examples) should be part of the https://github.com/grafana/grafana/blob/master/style_guides/frontend.md otherwise there will be a lot of overlaps. I think this can be short and having
> 
> > The pull request adds value and the impact of the change is in line with [Frontend Style Guide](https://github.com/grafana/grafana/blob/master/style_guides/frontend.md).
> 
> should cover all the code style checks without duplication. Otherwise I think it makes lots of sense.

I couldn't find any reference to magic strings or numbers in airBnb style guide so I left those in here.

**Comment by bergquist** [comment]
Reading this I think there is enough overlap when reviewing frontend and backend PR's. I think we should combine them and maybe have a small section for each that adds those extra things to consider for different parts of the codebase. What do you think?

**Comment by hugohaggmark** [comment]
> Reading this I think there is enough overlap when reviewing frontend and backend PR's. I think we should combine them and maybe have a small section for each that adds those extra things to consider for different parts of the codebase. What do you think?

I'm perfectly fine with that @bergquist

**Comment by xlson** [comment]
Great job!

I agree with @bergquist, there's enough overlap for us to add some backend specific things in there. Not sure if you want to update? Otherwise you could just rename the file and merge and we can update with any backend specific things in a separate PR.

File: style_guides/frontend-review-checklist.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+- [ ] The pull request closes one issue if possible and does not fix unrelated issues within the same pull request.
+- [ ] The pull request contains necessary tests.
+
+## Low level checks
+
+- [ ] The pull request contains a title that explains the PR.
+- [ ] The pull request contains necessary link(s) to issue(s).
+- [ ] The pull request contains commits with commit messages that are small and understandable.
+- [ ] The pull request does not contain magic strings or numbers that could be replaced with an `Enum` or `const` instead.
+- [ ] The pull request does not increase the number of `implicit any` errors.
```
Comment (dprokop) [inline style_guides/frontend-review-checklist.md]: As a follow up issue we could try to introduce some pre-commit check that would warn/error on increasing number of `implicity any` errors

File: style_guides/frontend-review-checklist.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+- [ ] The pull request does not contain uses of `any` or `{}` that are unexplainable.
```
Comment (dprokop) [inline style_guides/frontend-review-checklist.md]: In such cases we should strive for including comment that explains reasoning behind `any`/`{}` usage

File: style_guides/frontend-review-checklist.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+### Bug specific checks
+
+- [ ] The pull request contains only one commit if possible.
+- [ ] The pull request contains `closes: #Issue` or `fixes: #Issue` in pull request description.
+
+### Redux specific checks (skip if pull request does not contain Redux changes)
+
+- [ ] The pull request does not contain code that mutate state in reducers or thunks.
+- [ ] The pull request uses helpers `actionCreatorFactory` and `reducerFactory` instead of traditional `switch statement` reducers in Redux.
```
Comment (dprokop) [inline style_guides/frontend-review-checklist.md]: Yas! @hugohaggmark would you mind adding an md file later on with some docs about `actionCreatorFactory`/`reducerFactory` usage? I think the PR description you have created when introducing those would be a perfect fit for docs as it was very comprehensive.

**Comment by bergquist** [comment]
sounds like a good plan. move forward forward forward :)

**Review by dprokop — APPROVED** [review]
Like it @hugohaggmark!

**Review by xlson — APPROVED** [review]

File: style_guides/frontend-review-checklist.md (hunk 1 above)
Comment (hugohaggmark) [reply style_guides/frontend-review-checklist.md]: I think @bergquist had an excellent idea on how to solve this by breaking the CI-build if some type of errors reach a certain limit. But we should use that logic in pre-commit check as well.

File: style_guides/frontend-review-checklist.md (hunk 2 above)
Comment (hugohaggmark) [reply style_guides/frontend-review-checklist.md]: I changed the text somewhat, WDYT?

File: style_guides/frontend-review-checklist.md (hunk 3 above)
Comment (hugohaggmark) [reply style_guides/frontend-review-checklist.md]: Awesome suggestion, will do that!

**Pull request merged**